packages/react-chat/src/components/SystemResponse/ExtensionMessage.tsx

```diff
@@ -0,0 +1,40 @@
+import { Trace } from '@voiceflow/base-types';
+import { useEffect, useRef } from 'react';
+
+import { ResponseExtension } from '@/dtos/Extension.dto';
+
+import Message from '../Message';
+
+export interface ExtensionMessageProps {
+  extension: ResponseExtension;
+  trace: Trace.AnyTrace;
+}
+
+export const ExtensionMessage: React.FC<ExtensionMessageProps> = ({ extension, trace }) => {
+  const ref = useRef<HTMLSpanElement>(null);
+
+  useEffect(() => {
+    try {
+      // eslint-disable-next-line xss/no-mixed-html
+      const unmount = extension.render?.({ trace, element: ref.current as HTMLElement });
+      if (!unmount) return undefined;
+
+      return () => {
+        try {
+          unmount?.();
+        } catch (e) {
+          console.error(`Extension '${extension.name}' threw an error while unmounting: ${e}`);
```

Comment:
if they return something that is not a proper function, it's good we are catching this